frontend/src/lib/components/EditableField/EditableField.tsx

```diff
@@ -134,6 +145,15 @@ export function EditableField({
                                         onChange?.(e.target.value)
                                         setTentativeValue(e.target.value)
                                     }}
+                                    onBlur={
```

Comment:
This would address the user's direct issue, but it seems like quite unexpected behavior though – I wouldn't expect the new name to auto-save if I accidentally clicked outside of the text box. It seems the most intuitive thing would be if clicking "Save" on the insight also saved the tentative `EditableField` value, though that would require some complex `ref`s. Or… the field wouldn't have cancel/save in edit mode – then it'd get saved/canceled same way as all the rest of an insight.

Reply:
Less lazy version proposed :)

Reply:
This is a great approach but there's a problem we've brought onto ourselves: Because so many things rely on the whole `insight` reducer from `insightLogic`, every time `insight` changes EVERYTHING on the insight page re-renders, which is why fast typing results in huge lags:

![2022-08-31 16 18 03](https://user-images.githubusercontent.com/4550621/187706917-38957657-8bf0-4d95-9d9f-5e4456e4689e.gif)

Ideally we should reduce the reliance of the code on `insight`, because it's very rare that a single component cares about (for instance) both the name and calculation results. I'm not sure what the quick fix is (bearing in mind that we don't want `insightLogic` to be tangled further).